lib/cucumber/formatter/fail_fast.rb

```diff
@@ -0,0 +1,21 @@
+require 'cucumber/formatter/io'
+require 'cucumber/formatter/console'
+
+module Cucumber
+  module Formatter
+    class FailFast
+      def initialize(configuration)
+        @configuration = configuration
+      end
+
+      def after_test_case(test_case, result)
+        Cucumber.wants_to_quit = true unless result.ok? @configuration.strict?
```

Comment:
That because there are no test with a test case with an undefined step, then the test case result is "not ok" only in strict mode, and only then shall the execution be stopped.